src/OpenApi/src/Extensions/JsonObjectSchemaExtensions.cs

```diff
@@ -0,0 +1,279 @@
+        [typeof(decimal?)] = new() { Type = "number", Format = "double", Nullable = true },
+        [typeof(DateTime?)] = new() { Type = "string", Format = "date-time", Nullable = true },
+        [typeof(DateTimeOffset?)] = new() { Type = "string", Format = "date-time", Nullable = true },
+        [typeof(Guid?)] = new() { Type = "string", Format = "uuid", Nullable = true },
+        [typeof(char?)] = new() { Type = "string", Nullable = true },
+        [typeof(Uri)] = new() { Type = "string", Format = "uri" },
+        [typeof(string)] = new() { Type = "string" },
+    };
+
+    /// <summary>
+    /// Maps the given validation attributes to the target schema.
+    /// </summary>
+    /// <remarks>
+    /// OpenApi schema v3 supports the validation vocabulary supported by JSON Schema. Because the underlying
+    /// schema generator does not validation attributes to the validation vocabulary, we apply that mapping here.
+    ///
+    /// Note that this method targets <see cref="JsonObject"/> and not <see cref="OpenApiSchema"/> because it is
+    /// designed to be invoked via the `OnGenerated` callback provided by the underlying schema generator
+    /// so that attributes can be mapped to the properties associated with inputs and outputs to a given request.
+    ///
+    /// This implementation only supports mapping validation attributes that have an associated keyword in the
+    /// validation vocabulary.
+    /// </remarks>
+    /// <param name="schema">The <see cref="JsonObject"/> produced by the underlying schema generator.</param>
+    /// <param name="validationAttributes">A list of the validation attributes to apply.</param>
+    internal static void ApplyValidationAttributes(this JsonObject schema, IEnumerable<Attribute> validationAttributes)
+    {
+        foreach (var attribute in validationAttributes)
+        {
+            if (attribute is Base64StringAttribute)
+            {
+                schema["format"] = "byte";
+            }
+            if (attribute is RangeAttribute rangeAttribute)
+            {
+                schema["minimum"] = decimal.Parse(rangeAttribute.Minimum.ToString()!, CultureInfo.InvariantCulture);
+                schema["maximum"] = decimal.Parse(rangeAttribute.Maximum.ToString()!, CultureInfo.InvariantCulture);
+            }
+            if (attribute is RegularExpressionAttribute regularExpressionAttribute)
+            {
+                schema["pattern"] = regularExpressionAttribute.Pattern;
+            }
+            if (attribute is MaxLengthAttribute maxLengthAttribute)
+            {
+                schema["maxLength"] = maxLengthAttribute.Length;
+            }
+            if (attribute is MinLengthAttribute minLengthAttribute)
+            {
+                schema["minLength"] = minLengthAttribute.Length;
+            }
+            if (attribute is StringLengthAttribute stringLengthAttribute)
+            {
+                schema["minimum"] = stringLengthAttribute.MinimumLength;
```

Comment:
What's the difference between minimum and minLength?

Reply:
Oof -- good catch. This should be `minLength` and `maxLength` respectively. `minimum` is for representing minimum values for integers and `minLength` and `maxLength` are for the number of chars in a string or elements in an array.